src/main/pileup.js

```diff
@@ -80,6 +80,37 @@ function create(elOrId: string|Element, params: PileupParams): Pileup {
       ReactDOM.render(<Root referenceSource={referenceTrack.source}
                             tracks={vizTracks}
                             initialRange={params.range} />, el);
+
+  //if the element doesn't belong to document DOM observe DOM to detect
+  //when it's attached
+  var observer = null;
+  if (!document.contains(el)) {
+    observer = new MutationObserver(function(mutations) {
+      mutations.forEach(function(mutation) {
+        if (mutation.type === 'childList') {
+          var added= false;
+          for (var i=0; i<mutation.addedNodes.length;i++) {
+            if (mutation.addedNodes[i]===el) {
+              added = true;
+            }
+          }
+          if (added) {
+            if (reactElement) {
+              reactElement.setState({updateSize:true});
+            } else {
+              throw 'ReactElement was not initialized properly';
+            }
+          }
+        }
+      });
+    });
+    // configuration of the observer:
+    var config = { attributes: true, childList: true, characterData: true, subtree: true };
+
+    // start observing document
+    observer.observe(document, config);
```

Comment:
I have no experience with the `MutationObserver` API, but do you think observing the whole document DOM tree will result in some performance issues for us? Do you have an idea about how we can only watch some relevant elements and not the whole document?